resources/images/exif/exif.go

```diff
@@ -113,127 +113,104 @@ func (d *Decoder) Decode(r io.Reader) (ex *Exif, err error) {
+
 	var tm time.Time
 	var lat, long float64
 
-	if !d.noDate {
-		tm, _ = x.DateTime()
-	}
+	rootIfd := index.RootIfd
 
-	if !d.noLatLong {
-		lat, long, _ = x.LatLong()
-	}
+	ifd, err := rootIfd.ChildWithIfdPath(exifcommon.IfdGpsInfoStandardIfdIdentity)
+	if !d.noLatLong && err == nil {
+		gpsInfo, _ := ifd.GpsInfo()
 
-	walker := &exifWalker{x: x, vals: make(map[string]interface{}), includeMatcher: d.includeFieldsRe, excludeMatcher: d.excludeFieldsrRe}
-	if err = x.Walk(walker); err != nil {
-		return
+		lat = gpsInfo.Latitude.Decimal()
+		long = gpsInfo.Longitude.Decimal()
 	}
 
-	ex = &Exif{Lat: lat, Long: long, Date: tm, Tags: walker.vals}
+	exifIfd, _ := rootIfd.ChildWithIfdPath(exifcommon.IfdExifStandardIfdIdentity)
+	results, err := exifIfd.FindTagWithId(dateTimeTagId)
 
-	return
-}
+	if !d.noDate && err == nil && len(results) == 1 {
+		dateTimeTagEntry := results[0]
+		dateTimeRaw, _ := dateTimeTagEntry.Value()
+		dateTimeString := dateTimeRaw.(string)
 
-func decodeTag(x *_exif.Exif, f _exif.FieldName, t *tiff.Tag) (interface{}, error) {
-	switch t.Format() {
-	case tiff.StringVal, tiff.UndefVal:
-		s := nullString(t.Val)
-		if strings.Contains(string(f), "DateTime") {
-			if d, err := tryParseDate(x, s); err == nil {
-				return d, nil
-			}
-		}
-		return s, nil
-	case tiff.OtherVal:
-		return "unknown", nil
+		tm, _ = time.Parse(exifTimeLayout, dateTimeString)
 	}
 
-	var rv []interface{}
-
-	for i := 0; i < int(t.Count); i++ {
-		switch t.Format() {
-		case tiff.RatVal:
-			n, d, _ := t.Rat2(i)
-			rat := big.NewRat(n, d)
-			if n == 1 {
-				rv = append(rv, rat)
-			} else {
-				f, _ := rat.Float64()
-				rv = append(rv, f)
-			}
-
-		case tiff.FloatVal:
-			v, _ := t.Float(i)
-			rv = append(rv, v)
-		case tiff.IntVal:
-			v, _ := t.Int(i)
-			rv = append(rv, v)
-		}
+	walker := &exifWalker{index: &index, vals: make(map[string]interface{}), includeMatcher: d.includeFieldsRe, excludeMatcher: d.excludeFieldsrRe}
+	if err = walker.Walk(); err != nil {
+		return
 	}
 
-	if t.Count == 1 {
-		if len(rv) == 1 {
-			return rv[0], nil
-		}
-	}
+	ex = &Exif{Lat: lat, Long: long, Date: tm, Tags: walker.vals}
 
-	return rv, nil
+	return
 }
 
 // Code borrowed from exif.DateTime and adjusted.
-func tryParseDate(x *_exif.Exif, s string) (time.Time, error) {
+func tryParseDate(s string) (time.Time, error) {
 	dateStr := strings.TrimRight(s, "\x00")
 	// TODO(bep): look for timezone offset, GPS time, etc.
 	timeZone := time.Local
-	if tz, _ := x.TimeZone(); tz != nil {
-		timeZone = tz
-	}
 	return time.ParseInLocation(exifTimeLayout, dateStr, timeZone)
 }
 
+func processValue(valueRaw interface{}, tagType exifcommon.TagTypePrimitive, tagName string) (val interface{}, err error) {
+	switch tagType {
+	case exifcommon.TypeAscii:
+		s, isString := valueRaw.(string)
+		if isString && strings.Contains(tagName, "DateTime") {
+			if d, err := tryParseDate(s); err == nil {
+				return d, nil
+			}
+		}
+		return s, nil
```

Comment:
I'm not sure this could ever happen, but you do have an `isString` check above. If this is not a string at this point, `s` would be empty. So you should probably not return anything here.

Reply:
The rest of the good looks very good, but I ran this PR on a site I have that uses the `Exif`: 

https://github.com/bep/forsale

Where I now get this error:


```bash
exif" .>: error calling partial: "/Users/bep/dev/sites/forsale/layouts/partials/exif.html:3:6": execute of template failed: template: partials/exif.html:3:6: executing "partials/exif.html" at <.Exif>: error calling Exif: metadata init failed: no exif data
```

If there is no Exif data then `.Exif` should return nil.